app/src/main/java/ch/hikemate/app/ui/map/RunHikeScreen.kt

```diff
@@ -68,30 +67,80 @@ object RunHikeScreen {
 
 @Composable
 fun RunHikeScreen(
-    listOfHikeRoutesViewModel: ListOfHikeRoutesViewModel,
-    profileViewModel: ProfileViewModel,
+    hikesViewModel: HikesViewModel,
     navigationActions: NavigationActions,
 ) {
+  val selectedHike by hikesViewModel.selectedHike.collectAsState()
 
-  val context = LocalContext.current
-
-  val selectedRoute by listOfHikeRoutesViewModel.selectedHikeRoute.collectAsState()
-
-  LaunchedEffect(selectedRoute) {
-    if (selectedRoute == null) {
+  LaunchedEffect(selectedHike) {
+    if (selectedHike == null) {
       navigationActions.goBack()
     }
   }
-  val route = selectedRoute!!
 
-  // This will need to be changed when the "run" feature of this screen is implemented
-  val routeZoomLevel = MapUtils.calculateBestZoomLevel(route.bounds).toDouble()
+  // Can't display the details without a selected hike
+  if (selectedHike == null) {
```

Comment:
At least it is necessary for the compiler. I believe the `LaunchedEffect` is a separate thing as the main body of the function, so it might get executed afterwards. Hence, both checks are needed.

Reply:
I don't see why though since the `LaunchedEffect` will force the user to navigate back and thus it's technically impossible that this happens. 
I also don't see why this is true:
> At least it is necessary for the compiler.

Reply:
`LaunchedEffect(selectedHike) { ... }` only launches something, from what I understand we do not have any guarantee that the launched effect is executed first. It will be launched, not necessarily executed. This condition allows us to be certain 100% we do not have a null hike error.

Reply:
Plus, the `LaunchedEffect` does not stop the execution of the composable if the selected hike is null, it only calls `goBack`. That might take time to execute or something, at least the compiler is not happy with it, and I don't want to do an unsafe `!!`, so I check first.

Reply:
Ok then thanks for the explanation. That makes sense.